crates/runtime/src/memory.rs

```diff
@@ -84,18 +86,26 @@ impl MmapMemory {
             }
         } as usize;
         let minimum_bytes = minimum_pages.checked_mul(WASM_PAGE_SIZE as usize).unwrap();
-        let request_bytes = minimum_bytes.checked_add(offset_guard_bytes).unwrap();
+        let request_bytes = pre_guard_bytes
+            .checked_add(minimum_bytes)
+            .unwrap()
+            .checked_add(offset_guard_bytes)
+            .unwrap();
         let mapped_pages = plan.memory.minimum as usize;
         let mapped_bytes = mapped_pages * WASM_PAGE_SIZE as usize;
 
-        let mmap = WasmMmap {
-            alloc: Mmap::accessible_reserved(mapped_bytes, request_bytes)?,
+        let mut mmap = WasmMmap {
+            alloc: Mmap::accessible_reserved(0, request_bytes)?,
             size: plan.memory.minimum,
         };
+        if mapped_bytes > 0 {
+            mmap.alloc.make_accessible(pre_guard_bytes, mapped_bytes)?;
```

Comment:
Nit: this looks good, but while we're here can we rename `mapped_bytes` to `accessible_bytes` (seemingly incorrectly named prior to these changes)?